libraries/bot-applicationinsights/src/test/java/com/microsoft/bot/applicationinsights/TelemetryInitializerTests.java

```diff
@@ -0,0 +1,104 @@
+
+        TestAdapter testAdapter = new TestAdapter()
+        		.use(new TelemetryInitializerMiddleware(telemetryLoggerMiddleware, true));
+
+        // Act
+        // Default case logging Send/Receive Activities
+        TestFlow testFlow = new TestFlow(testAdapter, turnContext -> {
+            Activity typingActivity = new Activity(ActivityTypes.TYPING);
+            typingActivity.setRelatesTo(turnContext.getActivity().getRelatesTo());
+
+            turnContext.sendActivity(typingActivity).join();
+            try {
+                TimeUnit.MILLISECONDS.sleep(500);
+            } catch (InterruptedException e) {
+                // Empty error
+            }
+            turnContext.sendActivity(String.format("echo:%s", turnContext.getActivity().getText())).join();
+            return CompletableFuture.completedFuture(null);
+        })
+        .send("foo")
+            .assertReply(activity -> {
+                Assert.assertTrue(activity.isType(ActivityTypes.TYPING));
+            })
+            .assertReply("echo:foo")
+        .send("bar")
+            .assertReply(activity -> {
+                Assert.assertTrue(activity.isType(ActivityTypes.TYPING));
+            })
+            .assertReply("echo:bar")
+        .startTest().join();
+
+        verify(mockTelemetryClient, times(1));
+    }
+
+    @Test
+    public void telemetryInitializerMiddlewareNotLogActivitiesDisabled() {
+    	TelemetryClient mockTelemetryClient = Mockito.mock(BotTelemetryClient.class);
+
+        TelemetryLoggerMiddleware telemetryLoggerMiddleware = new TelemetryLoggerMiddleware(mockTelemetryClient, false);
+
+        TestAdapter testAdapter = new TestAdapter()
+        		.use(new TelemetryInitializerMiddleware(telemetryLoggerMiddleware, false));
+
+        // Act
+        // Default case logging Send/Receive Activities
+        TestFlow testFlow = new TestFlow(testAdapter, (turnContext) -> {
+            Activity typingActivity = new Activity(ActivityTypes.TYPING);
+            typingActivity.setRelatesTo(turnContext.getActivity().getRelatesTo());
+
+            turnContext.sendActivity(typingActivity).join();
+            try {
+                TimeUnit.MILLISECONDS.sleep(500);
+            } catch (InterruptedException e) {
+                // Empty error
+            }
+            turnContext.sendActivity(String.format("echo:%s", turnContext.getActivity().getText())).join();
+            return CompletableFuture.completedFuture(null);
+        })
+        .send("foo")
+            .assertReply(activity -> {
+                Assert.assertTrue(activity.isType(ActivityTypes.TYPING));
+            })
+            .assertReply("echo:foo")
+        .send("bar")
+            .assertReply(activity -> {
+                Assert.assertTrue(activity.isType(ActivityTypes.TYPING));
+            })
+            .assertReply("echo:bar")
+        .startTest().join();
+
+        verify(mockTelemetryClient, times(0));
```

Comment:
[Comment](https://github.com/microsoft/botbuilder-dotnet/blob/main/tests/integration/Microsoft.Bot.ApplicationInsights.Core.Tests/TelemetryInitializerTests.cs#L423)

Reply:
hope you don't mind I put `// Verify` instead of `// Assert` since we don't strictly have asserts here unlike in C#